### PR TITLE
#2363 add test for all_assets.json

### LINKED ISSUE #2363: Add test that checks all_assets.json for all common keys
## Abstract

We need to add a few more tests about the all_assets.json file. At least one that tests that the common keys are available in all assets. They can be missing and we will not find it out unless they are attempted to be used.

## Task

- We have a huge list of all of our supported assets in [this](https://github.com/rotki/rotki/blob/4f5292e0934d7eb375bdc42bb70754f6b44ecb1d/rotkehlchen/data/all_assets.json) file.
- Alll asset entries have a set of keys/attributes. This test is to go through each one of these entries just like [this](https://github.com/rotki/rotki/blob/4f5292e0934d7eb375bdc42bb70754f6b44ecb1d/rotkehlchen/tests/unit/test_assets.py#L139-L144) test does. The test should also be added in that same file.
- And for each entry check that the existing symbols are one of [these](https://github.com/rotki/rotki/blob/4f5292e0934d7eb375bdc42bb70754f6b44ecb1d/rotkehlchen/assets/resolver.py#L206-L221). More specifically check that:
    - All entries have the following keys:
        - symbol -> and it's a string
        - name -> and it's a string
        - type -> and it's a string
        - started -> and its an integer
    - Some entries optionally have:
        - active -> and it's boolean 
        - ended -> and it's an integer
        - forked -> and it's a string
        - swapped_for -> and it's a string
        - cryptocompare -> and it's a string
        - coingecko -> and it's astring
    - If type is ethereum_token so (AssetType.ETH_TOKEN or AssetType.ETH_TOKEN_AND_MORE):
        -  ethereum_address -> and it's a string
        - ethereum_token_decimals -> and it's an integer
  

If any of the above is not true fail the test with an informative error of what went wrong.

### PR DESCRIPTION
**Pull request by yabirgb**
This pull request closes #2363 There are some assets that are missing fields, for example `Australian Dollar`: `{'name': 'Australian Dollar', 'symbol': 'AUD', 'type': 'fiat'}
`

### CODE REVIEW

**Review by LefterisJP — COMMENTED** [review]
Hey @yabirgb thanks a lot for the PR!

Some small comments.

1. In your venv run `make lint` to see all linting errors that need fixing.
2. You mentioned that some assets are missing entries. Can you add them and then also update the md5 hash in the meta file and in the asset unit tests?
3. You will need to sign the CLA for the PR to be accepted. This essentially allows us to change the license if we need to do so in the future.
4. It would be beneficial for you to always create a new branch in your fork when you want to open a PR. This way you can work on multiple things in parallel and `develop` in your copy does not get extra commits.

**Comment by yabirgb** [comment]
@LefterisJP 
1. Sorry, forgot to pass the linter and had to leave for a moment. Already fixed this. 
2. Yes, should I open a different PR for this?
3. Done :)
4. Yes I usually do this. Do you have any naming convention or `issue-x` is fine?

**Comment by yabirgb** [comment]
Seems like the coins missing the `started` date are mostly fiat. What date should I use for this entries? (some were created before epoch time)

**Comment by LefterisJP** [comment]
> @LefterisJP Seems like the coins missing the `started` date are mostly fiat. What date should I use for this entries? (some were created before epoch time)

Oh wow ... indeed forget it about Fiat. This is on me then I should have described it better in the issue. Fiat does not need to have started date indeed.

**Comment by yabirgb** [comment]
> > @LefterisJP Seems like the coins missing the `started` date are mostly fiat. What date should I use for this entries? (some were created before epoch time)
> 
> Oh wow ... indeed forget it about Fiat. This is on me then I should have described it better in the issue. Fiat does not need to have started date indeed.

The non fiat assets that produce errors in the test are:

**Comment by LefterisJP** [comment]
Thanks that sounds good. For KFEE, it's a kraken internal asset. So it would make sense to put the launch date of Kraken: 1377993600

**Comment by LefterisJP** [comment]
Seems the test is failing at the moment.

**Comment by yabirgb** [comment]
> Seems the test is failing at the moment.

Yes, it is because of the fiat assets https://github.com/rotki/rotki/pull/2440/checks?check_run_id=1980449497#step:8:156 Tought you wanted to change those. Should I add some default values?

**Comment by LefterisJP** [comment]
No what I meant from the previous comment is that the test should ignored "started" key for Fiat assets as that is not useful/makes no sense there.

**Comment by yabirgb** [comment]
> No what I meant from the previous comment is that the test should ignored "started" key for Fiat assets as that is not useful/makes no sense there.

Okey, everything fixed now. Sorry, was my bad, I didn't get you on the first comment

File: rotkehlchen/tests/unit/test_assets.py [hunk 1]
```diff
@@ -144,6 +144,42 @@ def test_asset_identifiers_are_unique_all_lowercased():
         identifier_set.add(asset_id)
 
 
+def test_assets_have_common_keys():
+    """Test that checks that the keys and types of tokens in all_assets.json are correct"""
+    # Issue #2363
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_assets.py:149]: nitpick: no reason to add issue number here, nor the following token. Code is understandable.

File: rotkehlchen/tests/unit/test_assets.py [hunk 2]
```diff
@@ -144,6 +144,42 @@ def test_asset_identifiers_are_unique_all_lowercased():
...
+        asset_type = resolver.get_asset_data(asset_id).asset_type
+        if asset_type in (AssetType.ETH_TOKEN, AssetType.ETH_TOKEN_AND_MORE):
+            verify_against = verify_token
+        elif asset_type == AssetType.FIAT:
+            verify_against = verify_fiat
+
+        # make the basic checks
+        assert all((key in asset_data.keys() and isinstance(asset_data[key], key_type)
+                    for (key, key_type) in verify_against)), 'Key verification'\
+            f' in asset {asset_id} failed'
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_assets.py:174]: We never add `\` in our code to break lines. We favor readablity. The way to do it here would be: 


```suggestion
        msg = f'Key verification in asset {asset_id} failed'
        assert all((
            key in asset_data.keys() and isinstance(asset_data[key], key_type)
            for (key, key_type) in verify_against
         )), msg
```

Note: I may have a typo in the suggestion or something. Did not really run it to check.

File: rotkehlchen/tests/unit/test_assets.py [hunk 3]
```diff
@@ -144,6 +144,42 @@ def test_asset_identifiers_are_unique_all_lowercased():
...
+
+        # check the optional fields
+        for key, key_type in optional:
+            if asset_data.get(key) is not None:
+                assert isinstance(asset_data.get(key), key_type), 'Optional keys'\
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_assets.py:179]: same comment here. Move the assertion message to a new variable as per the suggestion above and split it in two lines

**Review by LefterisJP — COMMENTED** [review]
Looks good functionality-wise. Just some small comments.

**Review by LefterisJP — APPROVED** [review]
Thanks a lot! LGTM

**Pull request merged**